app/javascript/packages/password-strength/password-strength-element.ts

```diff
@@ -0,0 +1,188 @@
+import zxcvbn from 'zxcvbn';
+import { t } from '@18f/identity-i18n';
+import type { ZXCVBNResult, ZXCVBNScore } from 'zxcvbn';
+
+/**
+ * A mapping of zxcvbn scores to corresponding CSS BEM modifier classes.
+ */
+const PASSWORD_SCORE_CLASS_MODIFIERS: Record<ZXCVBNScore, string> = {
+  0: 'password-strength--very-weak',
+  1: 'password-strength--weak',
+  2: 'password-strength--average',
+  3: 'password-strength--good',
+  4: 'password-strength--great',
+};
+
+/**
+ * A mapping of zxcvbn scores to corresponding strength indicators.
+ */
+const PASSWORD_SCORE_STRENGTH: Record<ZXCVBNScore, string> = {
```

Comment:
WDYT of making one map of both of these combined? Minimizes the risk of them getting out of sync?

```js
interface PasswordScoreValue {
  string className;
  string strengthIndicator;
}
const PASSWORD_SCORE_VALUES: Record< ZXCVBNScore, PasswordScoreValue> = {
  0: {
    className: 'password-strength--very-weak',
    strengthIndicator: t('instructions.password.strength.i'),
  },
  // ... etc
}

// below
const allClassNames = Object.values(PASSWORD_SCORE_VALUES).map(({ className }) => className);

const { className, strengthIndicator } = PASSWORD_SCORE_VALUES[score];
```

Reply:
Technically TypeScript will keep us honest here since [`ZXCVBNScore` is typed as the set of all possible values](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/45226881424386a4ff80f797f64d00b628fa6b44/types/zxcvbn/index.d.ts#L58) and using it in the `Record` generic would produce an error if there were extra or missing keys, but I think it could make sense to consolidate this configuration, sure.

Reply:
Another idea could be to adapt the styles and strings to use the numeric scores, so that we don't need to have a mapping, e.g.

**Before:**

```scss
.password-strength--very-weak {
  // ...
}
```

```yml
password:
  strength:
    i: Very weak
    # ...
```

**After:**

```scss
.password-strength--score-0 {
  // ...
}
```

```yml
password:
  strength:
    0: Very weak
    # ...
```

Reply:
⬆️ That idea wouldn't work very well with the existing reset behavior:

https://github.com/18F/identity-idp/blob/1b11d1babc6e3bf2d49b55a243c75283f0ab5fff/app/javascript/packages/password-strength/password-strength-element.ts#L164

Maybe we could just assign it as an attribute instead, and style it that way:

```ts
this.addAttribute('score', score);
```

```scss
.password-strength[score="0"] {
  // ...
}
```

Reply:
the `score` attribute seems neat

Reply:
Updated in rebased 7093eb01a